File: update_leads_batch2.py
#!/usr/bin/env python3

import json
from datetime import datetime

def update_leadpool():
    # Load current data
    with open('src/data/leadpool.json', 'r') as f:
        data = json.load(f)
    
    # Define updates for batch 2
    updates = {
        "org-641": {  # Inkhouse
            "dataQuality": "verified",
            "stage": "qualified", 
            "score": 68,
            "isPipeline": True,
            "notes": ["Inkhouse - 20+ years, refurbished IT hardware provision. End of life IT services include data sanitization and asset remarketing. Mixed IT services provider."],
            "services": ["IT Hardware", "Data Sanitization", "End of Life IT"],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-645": {  # Interforce
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: Security guard company, not ITAD."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-649": {  # Inventory Source Limited
            "dataQuality": "rejected",
            "stage": "rejected", 
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: Equipment reseller with basic data wiping. No comprehensive ITAD facilities."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-652": {  # IT CONNECTS LONDON
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False, 
            "notes": ["REJECTED: Website under construction, cannot verify ITAD capabilities."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-653": {  # IT Links
            "dataQuality": "rejected", 
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: Dead website (404), unverifiable."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-654": {  # IT Parts and Spares
            "dataQuality": "rejected", 
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: Website inaccessible, cannot verify ITAD capabilities."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-663": {  # ITRegeneration
            "dataQuality": "rejected", 
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: Electronics surplus/used parts reseller, not ITAD processor."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        }
    }
    
    # Apply updates
    for item in data:
        if item["id"] in updates:
            update_data = updates[item["id"]]
            for key, value in update_data.items():
                item[key] = value
            print(f"Updated {item['id']}: {item['name']} -> {item['dataQuality']}")
    
    # Save updated data
    with open('src/data/leadpool.json', 'w') as f:
        json.dump(data, f, indent=2)
    
    print("Leadpool batch 2 updated successfully")

if __name__ == "__main__":
    update_leadpool()

File: update_leads_batch2.js
import fs from 'fs';

// Read the leadpool
const leadpool = JSON.parse(fs.readFileSync('src/data/leadpool.json', 'utf8'));

const updates = [
  {
    id: 'org-712',
    dataQuality: 'rejected',
    notes: ['REJECTED: NH Trading (NW) - computer export/wholesale business. Purchases and supplies ICT hardware globally but no evidence of ITAD processing facilities.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-719',
    dataQuality: 'rejected',
    notes: ['REJECTED: OCC Components - website not found (DNS error). Cannot verify ITAD operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-764',
    dataQuality: 'verified',
    notes: ['QUALIFIED: SecondLife - Complete ITAD services including asset valuation, lifecycle management, certified data wiping, IT recycling. Serves schools and businesses. London-based.'],
    stage: 'qualified',
    score: 73,
    isPipeline: true,
    services: ['IT Asset Valuation', 'Lifecycle Management', 'Data Wiping (Certified)', 'IT Recycling', 'Secure Disposal'],
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-765',
    dataQuality: 'rejected',
    notes: ['REJECTED: Secure Disk Technologies - website under construction. Cannot verify operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  }
];

// Apply updates
updates.forEach(update => {
  const index = leadpool.findIndex(lead => lead.id === update.id);
  if (index !== -1) {
    Object.assign(leadpool[index], update);
    leadpool[index].updatedAt = new Date().toISOString();
    console.log(`Updated ${update.id}: ${leadpool[index].name} - ${update.dataQuality}`);
  }
});

// Write back to file
fs.writeFileSync('src/data/leadpool.json', JSON.stringify(leadpool, null, 2));
console.log('Batch 2 updates complete');